src/benchmarks/micro/libraries/System.Numerics.Vectors/Perf_VectorConvert.cs

```diff
@@ -0,0 +1,231 @@
+        private static readonly float[] s_valuesFloat = ValuesGenerator.Array<float>(Vector<float>.Count * Iterations);
+        private static readonly byte[] s_valuesByte = ValuesGenerator.Array<byte>(Vector<byte>.Count * Iterations);
+        private static readonly sbyte[] s_valuesSByte = ValuesGenerator.Array<sbyte>(Vector<sbyte>.Count * Iterations);
+
+        // Vector.ConvertToXXX
+        [Benchmark]
+        public Vector<int> Convert_float_int() => Convert<float, int>(s_valuesFloat);
+
+        [Benchmark]
+        public Vector<uint> Convert_float_uint() => Convert<float, uint>(s_valuesFloat);
+
+        [Benchmark]
+        public Vector<long> Convert_double_long() => Convert<double, long>(s_valuesDouble);
+
+        [Benchmark]
+        public Vector<ulong> Convert_double_ulong() => Convert<double, ulong>(s_valuesDouble);
+
+        [Benchmark]
+        public Vector<float> Convert_int_float() => Convert<int, float>(s_valuesInt);
+
+        [Benchmark]
+        public Vector<float> Convert_uint_float() => Convert<uint, float>(s_valuesUInt);
+
+        [Benchmark]
+        public Vector<double> Convert_long_double() => Convert<long, double>(s_valuesLong);
+
+        [Benchmark]
+        public Vector<double> Convert_ulong_double() => Convert<ulong, double>(s_valuesULong);
+
+        // Vector.Narrow
+        [Benchmark]
+        public Vector<float> Narrow_double() => Narrow<double, float>(s_valuesDouble);
+
+        [Benchmark]
+        public Vector<sbyte> Narrow_short() => Narrow<short, sbyte>(s_valuesShort);
+
+        [Benchmark]
+        public Vector<byte> Narrow_ushort() => Narrow<ushort, byte>(s_valuesUShort);
+
+        [Benchmark]
+        public Vector<short> Narrow_int() => Narrow<int, short>(s_valuesInt);
+
+        [Benchmark]
+        public Vector<ushort> Narrow_uint() => Narrow<uint, ushort>(s_valuesUInt);
+
+        [Benchmark]
+        public Vector<int> Narrow_long() => Narrow<long, int>(s_valuesLong);
+
+        [Benchmark]
+        public Vector<uint> Narrow_ulong() => Narrow<ulong, uint>(s_valuesULong);
+
+        // Vector.Widen
+        [Benchmark]
+        public Vector<double> Widen_float() => Widen<float, double>(s_valuesFloat);
+
+        [Benchmark]
+        public Vector<short> Widen_sbyte() => Widen<sbyte, short>(s_valuesSByte);
+
+        [Benchmark]
+        public Vector<ushort> Widen_byte() => Widen<byte, ushort>(s_valuesByte);
+
+        [Benchmark]
+        public Vector<int> Widen_short() => Widen<short, int>(s_valuesShort);
+
+        [Benchmark]
+        public Vector<uint> Widen_ushort() => Widen<ushort, uint>(s_valuesUShort);
+
+        [Benchmark]
+        public Vector<long> Widen_int() => Widen<int, long>(s_valuesInt);
+
+        [Benchmark]
+        public Vector<ulong> Widen_uint() => Widen<uint, ulong>(s_valuesUInt);
+
+        private static Vector<TTo> Convert<TFrom, TTo>(TFrom[] values) where TFrom : struct where TTo : struct
```

Comment:
@adamsitnik, is there an issue on BDN that tracks adding better support for measuring operations that take less time than the timer frequency?
Likewise is there something that would more readily allow "random" inputs within some domain for those operations (for both primitive and structured data)?

Reply:
> is there an issue on BDN that tracks adding better support for measuring operations that take less time than the timer frequency?

I think that we are doing quite OK with things that take less time than the timer frequency, the problem might be benchmarks that take less than 1ns. Do you have any concrete example of an operation that is hard to benchmark properly as of today? We could create an issue and start a discussion with @AndreyAkinshin

BTW for this benchmark [OperationsPerInvoke](https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#operationsperinvoke) could be handy.

> Likewise is there something that would more readily allow "random" inputs within some domain for those operations (for both primitive and structured data)?

No. We rather expect the users to generate such data and put it into an array during a setup, and then just iterate over it during benchmark.